src/v/datalake/translation/partition_translator_v2.cc

```diff
@@ -0,0 +1,357 @@
+              "detected an offset range gap in [{}, {}), adjusting the begin "
+              "offset to avoid gaps in coordinator tracked offsets.",
+              begin_offset,
+              translation_result->start_offset);
+            translation_result->start_offset = begin_offset;
+        }
+
+        last_translated_offset = translation_result->last_offset;
+        auto checkpoint_result = co_await checkpoint_translation_result(
+          rcn, std::move(translation_result.value()));
+        if (checkpoint_result.errc != coordinator::errc::ok) {
+            vlog(
+              _logger.warn,
+              "Failed to checkpoint translated files: {}",
+              checkpoint_result);
+        }
+
+        reset_error = co_await _data_source->update_highest_translated_offset(
+          last_translated_offset,
+          translation_result->last_timestamp,
+          _term,
+          wait_timeout,
+          _as);
+        if (reset_error) {
+            vlog(
+              _logger.warn,
+              "error updating highest translated offset: {}",
+              reset_error);
+        }
+    }
+}
+
+const scheduling::translator_id& partition_translator::id() const {
+    return _data_source->ntp();
+}
+
+ss::future<> partition_translator::init(
+  scheduling::scheduling_notifications& scheduler,
+  scheduling::reservations_tracker& reservations) {
+    _scheduler = &scheduler;
+    _reservations = &reservations;
+    _initialized = true;
+    ssx::repeat_until_gate_closed_or_aborted(_gate, _as, [this] {
+        return ss::with_scheduling_group(_sg, [this] {
+            return translate_until_stopped().handle_exception(
+              [this](const std::exception_ptr& ex) {
+                  auto log_level = ssx::is_shutdown_exception(ex)
+                                     ? ss::log_level::debug
+                                     : ss::log_level::warn;
+                  vlogl(
+                    datalake_log,
+                    log_level,
+                    "[{}] Encountered exception in translation loop: {}",
+                    ex,
+                    _data_source->ntp());
+              });
+        });
+    });
+    return ss::make_ready_future();
+}
+
+ss::future<> partition_translator::close() noexcept {
+    _as.request_abort();
+    _ready_to_translate.broken();
+    if (_inflight_translation_state) {
+        _inflight_translation_state->as.request_abort();
+    }
+    _data_source->close();
+    return _gate.close();
+}
+
+scheduling::translation_status partition_translator::status() const {
+    // Biggest TODO: Currently we do not track intervals of target lag.
+    // So the scheduler is pretty much scheduling on defaults
+    // Track intervals return the correct remaining duration / memory
+    // reservations. The latter is already exposed as mux::buffered_bytes()
+    return {};
+}
+
+ss::future<std::chrono::milliseconds>
+partition_translator::current_lag_ms() const {
```

Comment:
i think this should take the highest offset stored in the log into account, otherwise for idle partitions the lag will always be growing with time

Reply:
yeah, i agree. not sure what the the function should be though 😕

Reply:
done